diesel/src/lib.rs

```diff
@@ -1,16 +1,95 @@
+//! [getting started guide]: https://diesel.rs/guides/getting-started/
+//! [many other long form guides]: https://diesel.rs/guides
 //!
-//! ## Crates
+//! # Where to find things
 //!
-//! * **Diesel**: core libary, documented here
-//! * [**Diesel CLI**](https://github.com/diesel-rs/diesel/tree/master/diesel_cli): command line tool that aids in managing your database schema
+//! ## Declaring your schema
 //!
+//! Diesel ensures that your queries are valid at compile time.
+//! This means it needs to know what your schema is.
+//! This is done using [the `table!` macro][`table!`].
+//! This macro is usually generated by `diesel print-schema` or [`infer_schema!`]
+//!
+//! [`table!`]: macro.table.html
+//! [`infer_schema!`]: FIXME: Host this crate on docs.diesel.rs because docs.rs won't render docs for this crate
+//!
+//! ## Getting started
+//!
+//! Queries usually start from either a table, or a function like [`update`].
+//! Those functions can be found [here](#functions).
+//!
+//! Diesel provides a [`prelude` module](prelude),
+//! which exports most of the most important traits and types.
+//! We are conservative about what goes in this module,
+//! and avoid anything which has a generic name.
+//! Files which use Diesel are expected to have `use diesel::prelude::*;`.
+//!
+//! [`update`]: fn.update.html
+//!
+//! ## Constructing a query
+//!
+//! Diesel breaks things into three categories:
+//!
+//! - "Query builder methods" are things that map to portions of a whole query
+//!   (such as `ORDER` and `WHERE`). These methods usually have the same name
+//!   as the SQL they map to, except for `WHERE` which is called `filter` in Diesel.
+//!   These methods live in [the `query_dsl` module](query_dsl).
+//! - "Expression methods" are things you would call on columns
+//!   or other individual values.
+//!   These methods live in [the `expression_methods` module](expression_methods)
+//!   You can often find these by thinking "what would this be called"
+//!   if it were a method
+//!   and typing that into the search bar
+//!   (e.g. `LIKE` is called `like` in Diesel).
+//!   Most operators are named based on the Rust function which maps to that
+//!   operator in `std::ops`
+//!   (For example `==` is called `.eq`, and `!=` is called `.ne`).
+//! - "Bare functions" are normal SQL functions
+//!   such as `sum`.
+//!   They live in [the `dsl` module](dsl).
+//!   Diesel only supports a very small number of these functions.
+//!   You can declare additional functions you want to use
+//!   with [the `sql_function!` macro][`sql_function!`].
+//!
+//! [`sql_function!`]: macro.sql_function.html
+//!
+//! ## Serializing and Deserializing
+//!
+//! Types which represent the result of a SQL query implement
+//! a trait called [`Queryable`].
+//!
+//! Diesel maps "Rust types" (e.g. `i32`) to and from "SQL types"
+//! (e.g. [`diesel::types::Integer`]).
+//! You can find all the types supported by Diesel in [the `types` module](types).
```

Comment:
Maybe also mention backend specific types?

Reply:
The docs for the `types` module shows links to backend specific types.